### ISSUE TITLE
[Accessibility] Default theme should at least have accessible coloring. 

### ISSUE THREAD

**Issue by Leopere** [ohmyform/ohmyform#54]
https://material.io/resources/color/#!/?view.left=1&view.right=0 readability needs to be possible almost universally ideally.  This means avoiding all caps strings in titles as screen readers read every letter and other nonsense like this however the first and foremost concern is just simply that we ship with a default that is visible for our sighted users who may have issues with color blindness and contrast issues.

**Comment by aitchiss** [comment]
I'd be happy to help with this! It will obviously need to involve changing some of your colours slightly - do you want me to just go for it and post a screenshot here so you can confirm you're still happy with the look?

**Comment by Leopere** [comment]
There is a WIP PR for this #40 where the future code base for this app resides, @wodka would likely merge whatever you work on there.

**Comment by Leopere** [comment]
@aitchiss have a look at this as well if you're looking for guidance https://github.com/ohmyform/ohmyform/issues/13 I'm not certain where we are exactly on this yet.

**Comment by aitchiss** [comment]
@Leopere thanks for all the guidance - that helped a lot.

On the home page I can see the WIP branch already has better color contrast. If we tweak the buttons a little more as below, we can clear all the contrast issues on these pages

![Screenshot 2019-10-10 at 18 18 53](https://user-images.githubusercontent.com/20773163/66591409-d00c6c00-eb8a-11e9-9c2b-ac035cf63060.png)

Let me know if that looks OK. 
Over the weekend I'm happy to dig a bit further as I see there are some other small tweaks that could be made for accessibility e.g. with landmarks and form labelling. Quick (possibly very silly) question though - are there test login credentials I can use to access beyond the sign up/register pages?

**Comment by Leopere** [comment]
Yes all of these things would be wonderful! Seeing as in the USA's Supreme Court just ruled that WCAG 2.0 is the minimum standard for websites on companies in the USA we would need to hit this  goal otherwise nobody in the US can use this. 

#### A huge digression
Europe's requirement of WCAG 2.1 is a disaster from what I hear from a friend of mine where apparently if the end user can change the font on your website and then it becomes inaccessible because of this then that means your site isn't WCAG 2.1 complaint.  Which to me sounds like a massive oversight, if the user can change the fonts on the site they clearly have opted to reject the websites design and the site should not be held accountable for that but Europe has exhibited a clear lack of understanding of how the world works in recent years despite their good intentions I suppose.  I could be hugely wrong here and am obviously interested in being proven wrong I always want to learn how not to suck in the future but anyways I don't know if we need to aim for WCAG 2.1 but if we can it would obviously be nice.

**Comment by Leopere** [comment]
# tl;dr
Yes it looks good thank you!  We're looking forward to any further contributions!  Feel free to support us further by joining the Discord or starring the repository.  http://ohmyform.com/contact/

**Comment by aitchiss** [comment]
EU only requires 2.0 under the Equality Act - I'd be really interested to hear more about what your friend said as it's very different than my experience! The Equality Act simply requires that everyone can access your online service, which is commonly accepted as WCAG 2.0.

I've been following the Domino's case really closely - it's great news for accessibility they declined to hear their case and it certainly sets a precedent for other law suits against online service providers.

I guess to introduce myself more I should say I'm really into web accessibility 🤓 I maintain www.upyoura11y.com

I'm looking to get more experience conducting accessibility audits and happy to help you folks with this a little bit!

**Comment by Leopere** [comment]
Great to hear!  I can ask him to chime in on this thread if you wish, we'll see what he has to elaborate on the topic.  That said we're hoping in the future to have a proper consultancy corp next to this repository allowing people to get work one of these days through us as certifiable OMF contributors seeing as its [AGPL 3.0](https://github.com/ohmyform/ohmyform/blob/master/SUBLICENSE.md) on the rewrite.  I actually wonder if we will be able to strip the MIT license soon because I want people to trust that we're not going to just up and vanish with this one day.

**Comment by Leopere** [comment]
He just linked me this stuff.

`AA is a WCAG standard`
`https://www.w3.org/WAI/news/2018-09-13/WCAG-21-EN301549/`

**Comment by hathawayweb** [comment]
Hi, I am the friend he keeps mentioning.

I originally had heard about the WCAG 2.1 Standards being adopted by Europe was from a member who writes the technical standard for WCAG. I then had read articles, including what I found below. The standard must be followed in regards to ICT products/services. If I am mistaken by this, I do apologize. This is just from my experience, as well as heard and read. 

https://www.w3.org/WAI/news/2018-09-13/WCAG-21-EN301549/
https://www.w3.org/blog/2018/09/wcag-2-1-adoption-in-europe/
https://siteimprove.com/en-us/accessibility/eu-web-accessibility-directive/
https://www.etsi.org/deliver/etsi_en/301500_301599/301549/02.01.02_60/en_301549v020102p.pdf

As for the Dominos case, it is very interesting. The Dominos case sets a precedent really for ADA Compliance and WCAG (as the person filled with the desire for them to follow WCAG). Also sets a precedent on other federally/state ruled cases such as Winn-Dixie, Beyonce, etc. Just a continuation of the already present standards for public organizations throughout the states, which will just continue growing.

https://chainstoreage.com/operations/winn-dixie-case-puts-spotlight-website-accessibilitycompliance
https://www.boia.org/blog/beyonces-website-the-focus-of-an-accessibility-lawsuit

**Comment by aitchiss** [comment]
Thanks for sharing! You're definitely right - 2.1 is the standard, but you only need to meet the AA level and not the full AAA (thank goodness as AAA is so strict!)

I'm based in Europe so I totally wasn't aware that US didn't require (so far) things like being able to resize text. So this chat has been really interesting! I think you guys can safely aim for the 2.1 AA standard though; in my experience it is not so difficult as long as standard semantic html practices are used 🙂

Thanks for sharing and letting me hijack this thread into a chat! 😂

**Comment by Leopere** [comment]
I feel like opportunities for learning are part of the spirit of open source software it was less of a hijack as it was a public contextualization of the importance of the goals in this issue!  Thanks for indulging us in this exploration.

**Comment by aitchiss** [comment]
I've put up a PR 🙂

It covers fixes on the login, register and recover pages and some fixes that help issues across the app (e.g. adding the language tag). Hopefully this helps, and gets the ball rolling on any further accessibility fixes.

Let me know if you've got any questions about any of it!
